### PR TITLE
[vSphere] Add unique document id for all the data-streams

### PR DESCRIPTION
**Pull request by kush-elastic**
- Enhancement

## Description
Add unique document `id` for all the data-streams

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [x] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## Related issues
- Relates https://github.com/elastic/obs-integration-team/issues/73

### CODE REVIEW

File: packages/vsphere/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.16.3-next"
+  changes:
+    - description: Added the id field to all data streams for improved identification and tracking.
```
Comment (harnish-elastic) [inline packages/vsphere/changelog.yml:4]:
```suggestion
    - description: Add the id field to all data streams to improve identification and tracking.
```

File: packages/vsphere/data_stream/host/sample_event.json [hunk 2]
```diff
@@ -74,26 +71,6 @@
                     "pct": 0.015
                 }
             },
-            "datastore": {
```
Comment (harnish-elastic) [inline packages/vsphere/data_stream/host/sample_event.json:77]: Can you re-run the system tests and generate this file? I think these shouldn't be remove!

File: packages/vsphere/data_stream/resourcepool/sample_event.json [hunk 3]
```diff
@@ -62,9 +64,8 @@
     ],
     "vsphere": {
         "resourcepool": {
-            "fingerprint": "IqGup33ooQgbWIgSs7+sAcX4MDo=",
```
Comment (harnish-elastic) [inline packages/vsphere/data_stream/resourcepool/sample_event.json:65]: I can see the fingerprint has removed but the id field is not generated. Can you generate the system tests?

File: packages/vsphere/manifest.yml [hunk 4]
```diff
@@ -9,7 +9,7 @@ categories:
   - virtualization
 conditions:
   kibana:
-    version: "^8.15.2"
+    version: "^8.16.2"
```
Comment (harnish-elastic) [inline packages/vsphere/manifest.yml:12]: I think the beats id changes will be available from `8.17.0` version. So can you change this to `8.17.0`?

File: packages/vsphere/data_stream/virtualmachine/fields/fields.yml [hunk 5]
```diff
@@ -19,16 +19,22 @@
...
       # Reason to add as a dimension field: Each VM has Unique name assigned.
       dimension: true
       description: >
-        Virtual Machine name.
+        Virtual machine name.
     - name: os
       type: keyword
       description: >
-        Virtual Machine Operating System name.
+        Virtual machine Operating System name.
```
Comment (harnish-elastic) [inline packages/vsphere/data_stream/virtualmachine/fields/fields.yml:37]: Can we please follow first letter capital and rest small across the fields.yml?

File: packages/vsphere/data_stream/resourcepool/sample_event.json (hunk 3 above)
Comment (kush-elastic) [reply packages/vsphere/data_stream/resourcepool/sample_event.json:65]: Updated

File: packages/vsphere/data_stream/host/sample_event.json (hunk 2 above)
Comment (kush-elastic) [reply packages/vsphere/data_stream/host/sample_event.json:77]: Updated

File: packages/vsphere/manifest.yml (hunk 4 above)
Comment (kush-elastic) [reply packages/vsphere/manifest.yml:12]: We are back-porting the ID changes to 8.16.2 onwards.
PR link: https://github.com/elastic/beats/pull/41700

File: packages/vsphere/data_stream/virtualmachine/fields/fields.yml (hunk 5 above)
Comment (kush-elastic) [reply packages/vsphere/data_stream/virtualmachine/fields/fields.yml]: updated

**Review by harnish-elastic — APPROVED** [review]

File: packages/vsphere/data_stream/cluster/fields/fields.yml [hunk 6]
```diff
@@ -24,23 +24,29 @@
...
+            Number of hosts associated with the cluster.
         - name: names
           type: keyword
           description: >
             List of all the host names associated with the cluster.
+    - name: id
+      type: keyword
+      description: >
+        Unique cluster ID.
+      # Reason to add as a dimension field: to uniquely identify the cluster ID.
```
Comment (ishleenk17) [inline packages/vsphere/data_stream/cluster/fields/fields.yml:48]:
```suggestion
      # Added as a dimension field, to uniquely identify the cluster ID.
```

File: packages/vsphere/data_stream/datastore/fields/fields.yml [hunk 7]
```diff
@@ -67,6 +67,12 @@
       type: keyword
       description: >
         List of all the host names associated with the datastore.
+    - name: id
+      type: keyword
+      description: >
+        Unique datastore ID.
+      # Reason to add as a dimension field: to uniquely identify the datastore ID.
```
Comment (ishleenk17) [inline packages/vsphere/data_stream/datastore/fields/fields.yml:74]: ame change as above

**Comment by muthu-mps** [comment]
Fixes #11849, #11850, #11851, #11852, #11853, #11854, #11855

**Review by lucian-ioan — APPROVED** [review]

**Review by ishleenk17 — APPROVED** [review]

**Pull request merged**